### PR TITLE
feat: Gumbel-Softmax and Activation Interface layers

### LINKED ISSUE #123: [FEAT] Create an interface layer connecting categorical distributions to specific Gumbel-Softmax layers
### Interface layer implementation
To process the logits outputs of a Generator correctly, information about the type of features being synthesized is necessary.
The idea of this layer is to leverage information available in the DataProcessor of the synthesizer in order to create Gumbel-Softmax activations for each of the categorical distributions, TanH/ReLU activations for other features and concatenating all outputs in the original order.
In the end this layer should be initialized leveraging just attributes of the processor and work as a model component like any other TF/Keras layer.

Requirements:
- TF implementation
- Gumbel-Softmax layer implementation (see below)
- Tests

### Gumbel-Softmax layer implementation
A Gumbel-Softmax layer works on logits or probability outputs of a model over the distribution of a categorical feature.
The required computations of this layer's forward method are the following:
- Soft sample: Softmax of a tensor consisting of a Gumbel sample added to logits. The gradients of this computation are stored.
- Hard sample: A one-hot encoded categorical variable with a stochastically sampled class activated. The gradients of this computation are ***not*** stored, but used only for obtaining _real_ categorical samples.

Note: Generally only the hard sample should be returned in the forward call, alternatively both can be returned in an (hard, soft) tuple.

Requirements:
- TF implementation
- A standalone custom layer implemented in utils that produces the requested output assuming as input logits from a single categorical distribution
- Tests

### PR DESCRIPTION
**Pull request by jfsantos-ds**
Adds Gumbel-Softmax layer and an interface layer that leverages a data object schema to instantiate appropriate activation functions (Gumbel-Softmax or TanH).
The categorical part of the output of the GS layer is a "sensible" one hot vector instead of a tensor of floats.
The output of the layer can also be applied directly on the inverse_transform method of a data processor.
Adds a schema object to the base processor that can be leveraged for easier instantiation of the above implementations.
Test suites added for both implementations.
Closes #123

### CODE REVIEW

**Review by fabclmnt — APPROVED** [review]
🚀
